lib/src/components/__tests__/SimpleComponents.test.tsx

```diff
@@ -337,3 +338,15 @@ describe('NxTableContainer.Footer', function() {
     expect(shallow(<NxTableContainer.Footer/>)).toMatchSelector('div.nx-table-container__footer');
   });
 });
+
+describe('NxPageLocalHeader', function() {
+  it('makes an <aside> with an nx-page-local-header class', function() {
+    expect(shallow(<NxPageLocalHeader/>)).toMatchSelector('aside.nx-page-local-header');
```

Comment:
Would you consider `header`, `section` or even `div` (if it is just for layout)  instead of an `aside` element?

> The HTML \<aside\> element represents a portion of a document whose content is only indirectly related to the document's main content. [[mdn]](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside)

> The \<aside\> element is used to identify content that is related to the primary content of the webpage, but does not constitute the primary content of the page. Author information, related links, related content, and advertisements are examples of content that may be found in an aside element. [[html.com]](https://html.com/tags/aside/)

Our primary use would be global Nav options, not generally related to the main content.

Reply:
That could've actually references https://github.com/sonatype/sonatype-react-shared-components/pull/372/files#diff-ff772c2424b5566111a42aca9953a7e851d8d341d9fea979605a3e706f904c56R79 instead of line 344, but I didn't recognize the change in `SimpleComponents.tsx` for what it was

Reply:
> Would you consider header, section or even div (if it is just for layout) instead of an aside element?

Could you clarify in what way this header doesn't seem like a good fit for `aside`?  It contains a back button to another page and the sort of global navigations/buttons that exist in the top-right corner of the IQ UI.  To me those seem like a fit for the purpose of `aside`.

> Our primary use would be global Nav options

Right, the global sidebar is also an `aside`.

Reply:
The `aside` element is supposed to be used for supplementary content related to, if tangentially, the main content.

Our uses of the aside are not related to the main content, nor are they really content.

We have what probably should be a `section` in the case of the global side-nav with nav element for the list of "main nav links" in that, and possibly `header` or `div` element for the "header" section and a nav for the "utility nav links" menus.  The back button / breadcrumbs could also be in `nav` elements in the `header` or not.

The "side nav" and "main header" appear on every page so they are not really related to the main content, so aside seems inappropriate. The html.com link above 

That all being said, semantic elements are great and all, but sometimes the minutia of it, and especially an not-well-defined always kinda-iffy element like `aside`, is not worth debating about, so I wouldn't consider it a blocker, more of a request, or even request for discussion.

Reply:
I think that this is an appropriate use of `<aside>`. 

W3C says: "Example of contents: pull quotes, **sidebars, advertising**, **groups of nav elements**, etc. Other content that is considered separate from the main content of the page." Emphasis mine.

MDN describes it likewise, "The HTML <aside> element represents a portion of a document whose content is only **indirectly related to the document's main content**. Asides are frequently presented as sidebars or call-out boxes." again emphasis mine.

Reply:
Right I still think this is a correct use of `aside` after re-reading the specs and docs posted in this thread.  If ads are relevant enough to be an aside, then surely this header is.  And at the same time I don't believe this header is part of the "main content" of the page.

> The "side nav" and "main header" appear on every page so they are not really related to the main content, so aside seems inappropriate.

I think it is actually extremely common for `aside`s to be things that appear on every page.

> nor are they really content.

What's your definition of content?  Everything rendered on the page is content I think.

> We have what probably should be a section in the case of the global side-nav with nav element for the list of "main nav links" in that

FWIW while the global sidebar itself is an `aside`, the nav links within are a `nav`.  

> and possibly header or div element for the "header" section 

From WHATWG on `header` elements

> The header element represents a group of introductory or navigational aids.

> A header element is intended to usually contain the section's heading ..., but this is not required.

Hmm so I'm now convinced that it would also be valid as a `header` due to the mention of "navigational aids".  @BigAB would that be your preference? @glenhunter what do you think?

> and a nav for the "utility nav links" menus

I hesitate to insist on that element being a `<nav>` because I'm not confident that everything in it will be navigation.  For instance in IQ that includes the notifications menu.

Reply:
I wouldn't have any issues with it being a `header`, that _might_ make more sense than `aside` but it might be a choice based on layout rather than content.

I do not think that it should be `nav`.

Reply:
> I wouldn't have any issues with it being a header, that might make more sense than aside but it might be a choice based on layout rather than content.

I think I'm going to stick with `aside`

Reply:
I am totally fine with sticking to `aside` but just for furthering the conversation and to address the sidebar thing, please have a look at the **Incorrect use of \<aside\>** section in this article: http://html5doctor.com/understanding-aside/

And actually the whole article as well.

Reply:
> I am totally fine with sticking to `aside` but just for furthering the conversation and to address the sidebar thing, please have a look at the **Incorrect use of <aside>** section in this article: http://html5doctor.com/understanding-aside/
> 
> And actually the whole article as well.

From the [updated article](http://html5doctor.com/aside-revisited/):

> It did not appear to be an appropriate element for holding secondary content related to the site as a whole, typically known as a “sidebar”. From the comments here and elsewhere, you can see this definition was not accepted by developers. The spec writers listened, and aside is now acceptable for secondary content when not nested within an article element.

Reply:
Interesting but I don't think that's relevant to what we are talking about.

Reply:
> Interesting but I don't think that's relevant to what we are talking about.

What do you mean it isn't relevant? we're discussing wether or not this kind of secondary content is valid as an `<aside>` element. Your article just pointed out it is.

Reply:
The correction and the thing you highlighted is specifically about using an **`aside`** element **within an article element**. Which was part of the original article and then removed.

Reply:
> The correction and the thing you highlighted is specifically about using an **`aside`** element **within an article element**. Which was part of the original article and then removed.

"aside is now acceptable for secondary content when **not** nested within an article element."
AFAIK we're not using `<article>` — which makes that point relevant to our current discussion.

Reply:
> Your article just pointed out it is.

If you read the **Conclusion** part of the updated article it has:

> For content that is not the primary focus of an article (or page) **but is still related to the article (or page)**, aside is what you need, regardless of its visual design.

Which is where my mind stays, if the aside is not related to the main content, and it is just a global nav and menus on every page, is it the proper element?

Reply:
> Which is where my mind stays, if the aside is not related to the main content, and it is just a global nav and menus on every page, is it the proper element?

The more clear quote from the updated article:

> When used within an article element, the contents should be specifically related to that article (e.g., a glossary). When used outside of an article element, the contents should be related to the site (e.g., a blogroll, groups of additional navigation, and even advertising if that content is related to the page).

Further, see the third example here for a more canonical source: https://html.spec.whatwg.org/multipage/sections.html#the-aside-element

Reply:
I see those all as supporting my point.

**From the "more clear quote":**
> if that content is related to the page

**From the whatwg spec:**
> consists of content that is tangentially related to the content

> this aside contains two sections that are tangentially related
  to the page, namely, links to other blogs, and links to blog posts
  from this blog
  
The H1 heading of that page being `<h1>My wonderful blog</h1>` and the content being a list of blog posts.


#### Content
As for my definition of **content**, it would be any images, videos, tables of information or prose that relays information on a topic.

I would consider navs, buttons, and inputs as **controls**, not content but rather a way to interact with the data or site as a whole. There are exceptions of course, for example a list of related links could be nav, but it is also content in the sense that it is showing titles of other articles on the content topic.

I think the unfortunate use of "side-bar" in the spec and therefore related blog articles, is the main cause of any confusion, because they don't mean what we traditionally in web think of as a sidebar but more:

>side·bar /ˈsīdˌbär/
> a short article in a newspaper or magazine, typically boxed, placed alongside a main article, and containing additional or explanatory material.

This has been a very good discussion, but please don't hold anything up because of it. Use `aside`, there will be literally no harm done by using it, but I do think that asides are meant for content with a relation